### PR TITLE
use ci.bokeh.org for artifact storage

### PR DESCRIPTION
**Pull request by bryevdv**
Migrate CI test artifact storage to ci.bokeh.org

### CODE REVIEW

**Comment by bryevdv** [comment]
@mattpap please review this as it touches your examples code. I thought there was actually still an issue to fix but to my surprise this build succeed, and everything (e.g. the report) looks OK. But it would be good to get your input. 

Once this is merged, we can delete the Anaconda s3 bucket for build artifacts. Please confirm my understanding that the `bokeh-scala` bucket can also be deleted at that time.

File: bokeh/_testing/util/examples.py [hunk 1]
```diff
@@ -265,11 +237,11 @@ def store_img(self, img_data):
...
         if isfile(self.img_path):
-            trace("%s Uploading image to S3 to %s" % (green(">>>"), self.img_url_path))
-            upload_file_to_s3(self.img_path, self.img_url_path, "image/png")
+            trace("%s Uploading image to S3 to %s" % (green(">>>"), self.img_path))
+            upload_file_to_s3_by_job_id(self.img_path, "image/png")
         if isfile(self.diff_path):
-            trace("%s Uploading image to S3 to %s" % (green(">>>"), self.diff_url_path))
-            upload_file_to_s3(self.diff_path, self.diff_url_path, "image/png")
+            trace("%s Uploading image to S3 to %s" % (green(">>>"), self.diff_path))
+            upload_file_to_s3_by_job_id(self.diff_path, "image/png")
```
Comment (bryevdv) [inline bokeh/_testing/util/examples.py]: I believe the paths here are the local file paths, which is why the uploads are not happening correctly

**Comment by mattpap** [comment]
> (...) that the `bokeh-scala` bucket can also be deleted at that time.

Yes. It should actually be empty.

File: bokeh/_testing/util/examples.py (hunk 1 above)
Comment (mattpap) [reply bokeh/_testing/util/examples.py]: I think the correct approach is:
```
upload_file_to_s3(self.diff_path, join(JOB_ID, self.diff_url_path), "image/png")
```
The examples' path situation is a bit messy and exacerbated by allowing for having reports with remote (uploaded) images or local images (which is useful when running tests locally).

**Comment by bryevdv** [comment]
@mattpap this was happening before, then stopped, and is now happening again. Basically, all the tests just fail with no explanation:
```
tests/test_examples.py::test_file_examples[/home/travis/build/bokeh/bokeh/examples/models/file/iris.py-example16-config16] 
[INFO] Example run in 0.651s
[INFO] Example rendered in 0.770s
[JS] {log} file:///home/travis/miniconda/envs/testenv/lib/python3.7/site-packages/bokeh/server/static/js/bokeh.js:3489:21 [bokeh] setting log level to: 'info'
[JS] {info} file:///home/travis/miniconda/envs/testenv/lib/python3.7/site-packages/bokeh/server/static/js/bokeh.js:7621:34 [bokeh] document idle at 26 ms
FAILED
```
Tried looking at the `examples.log` before and it seemed to be heavily truncated and not useful.

**Comment by bryevdv** [comment]
Hrm I guess `ref_url` was missing, not sure why the linter did not capture that.

**Comment by mattpap** [comment]
Bare `FAILED` usually means an internal failure. There should be a summary at the end (one line) that should give a hint what's wrong with each test. I suppose we are using py.test improperly, but given the amount of magic there, it's understandable.

**Comment by bryevdv** [comment]
@mattpap OK things are passing now, and I see the images in S3. I don't see images in the HTML report, but I think this is a pre-exisiting issue (I have not seen images in the report for awhile). Do you mind if I merge this now?

FYI the URL scheme for the images is e.g. 

https://s3.amazonaws.com/ci.bokeh.org/travis/495031488/1.1.0dev4-048cc46/examples/app/apply_theme.png

**Comment by bryevdv** [comment]
Tho I just realized, do thew ref images need to not be stored under the job id? else how are they located by future jobs.

**Comment by bryevdv** [comment]
Could also move `builds` to `travis/builds` and test images saved under `travis/tests` or `travis/tests/ref_images`, anticipating having other build systems dump results, etc. in this bucket

**Comment by mattpap** [comment]
> OK things are passing now, and I see the images in S3. I don't see images in the HTML report, but I think this is a pre-exisiting issue (I have not seen images in the report for awhile).

The report is clearly wrong, e.g. `src=(unknown)` for all current images, which doesn't happen in `master`. Normally you don't see any images, because they are hidden, so that only what matters (i.e. what failed) is visible. I will look into this tomorrow, because I need reliable image diff reporting to get my work done.

**Comment by bryevdv** [comment]
Great, I will wait for your fixes to this branch.

**Comment by mattpap** [comment]
I fixed the report (too many methods were removed from `Example` class). However, to make sure that everything works correctly, the following steps are necessary:

1. merge this PR
2. wait for Travis CI to run tests on `master` and upload images
3. rebase another PR on top of `master` and rerun tests
4. make sure there are no "reference image doesn't exist` warnings

**Comment by mattpap** [comment]
It works locally, but not on Travis CI.

**Comment by bryevdv** [comment]
@mattpap is it ready to merge after your last commit?

**Comment by mattpap** [comment]
Now it says:

GET https://ci.bokeh.org/1.1.0dev6-49246cc/bokehjs/examples/anscombe/anscombe.png 404

(similarly for all other examples; this is the current image).

**Comment by mattpap** [comment]
The reported Bokeh version is `1.1.0dev6+11.g49246cc`, though I'm not sure why, because such commit doesn't exist.

**Comment by mattpap** [comment]
Locally it reports correctly `1.1.0dev4-11-g8053ea7`.

**Comment by mattpap** [comment]
Actually this is irrelevant.

**Comment by mattpap** [comment]
@bryevdv, now it works.

**Comment by bryevdv** [comment]
OK I wil l merge this now. I have to step out for 30 minutes but I will rebase or make a new PR when I get back if you haven't already.

**Pull request merged**